src/cdk-experimental/popover-edit/edit-event-dispatcher.ts

```diff
@@ -37,13 +68,130 @@ export class EditEventDispatcher {
   }
   private _editRef: EditRef<any>|null = null;
 
+  // Optimization: Precompute common pipeable operators used per row/cell.
+  private readonly _distinctUntilChanged =
```

Comment:
Could you do the types like this instead?
```ts
private readonly _distinctUntilChanged: MonoTypeOperatorFunction<Element|null> =
    distinctUntilChanged<Element|HoverContentState|boolean|null>();
```

That should avoid needing to use `as`

Reply:
The reason for the as elsewhere is that the instance is being shared between circumstances with different subsets of the possible types. I don't think this would work around that.